sqllogictest/src/runner.rs

```diff
@@ -111,29 +111,47 @@ impl TestError {
     }
 }
 
+#[derive(Debug, Clone)]
+pub enum RecordKind {
+    Statement,
+    Query,
+}
+
+impl std::fmt::Display for RecordKind {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            RecordKind::Statement => write!(f, "statement"),
+            RecordKind::Query => write!(f, "query"),
+        }
+    }
+}
+
 /// The error kind for running sqllogictest.
 #[derive(thiserror::Error, Debug, Clone)]
 pub enum TestErrorKind {
     #[error("parse error: {0}")]
     ParseError(ParseErrorKind),
-    #[error("statement is expected to fail, but actually succeed:\n[SQL] {sql}")]
-    StatementOk { sql: String },
-    #[error("statement failed: {err}\n[SQL] {sql}")]
-    StatementFail {
+    #[error("{kind} is expected to fail, but actually succeed:\n[SQL] {sql}")]
+    Ok { sql: String, kind: RecordKind },
+    #[error("{kind} failed: {err}\n[SQL] {sql}")]
+    Fail {
         sql: String,
         err: Arc<dyn std::error::Error + Send + Sync>,
+        kind: RecordKind,
+    },
+    #[error("{kind} is expected to fail with error:\n\t\x1b[91m{expected_err}\x1b[0m\nbut got error:\n\t\x1b[91m{err}\x1b[0m\n[SQL] {sql}")]
```

Comment:
would be better if this can be controlled by flags, otherwise will mess up people who pipe the output to a file.

Reply:
Indeed. A similar problem: https://github.com/risinglightdb/sqllogictest-rs/issues/104

Reply:
fix later